src/dotnet-core-uninstall/Shared/BundleInfo/Bundle.cs

```diff
@@ -11,6 +11,7 @@ internal abstract class Bundle : IEquatable<Bundle>
         public BundleArch Arch { get; }
         public string UninstallCommand { get; }
         public string DisplayName { get; }
+        public bool UninstallAllowed { get; set; }
```

Comment:
Add this field in the class is not ideal. One code smell is it breaks encapsulation -- a big chuck of the operation logic on this field is in a different class VisualStudioSafeVersionsExtracter.

There are 2 way to go around it.

1. Remove this field. And simply run the VisualStudioSafeVersionsExtracter logic every time we need it. The downside is perf and duplication, we may need to compute twice. However, I think this is fine, considering this is a pure compute and we don't really have a time to compute it twice in one process live time. I think it fits this better, since the current code is very functional style.

2. Move VisualStudioSafeVersionsExtracter to Bundle. This is more of an "OOP" way of doing it. However, that would make Bundle class bigger. And currently non of the filtering operation is on Bundle class. It is odd for this code base.